### ISSUE TITLE
Update Ubuntu (and therefore Qemu) Version

### ISSUE THREAD

**Issue by Alexhuszagh** [cross-rs/cross#557]
Currently, on images provided by cross v0.2.1, we get the following after loading into the image:

# Cross Information:

```bash
$ cross --version
cross 0.2.1
cargo 1.51.0 (43b129a20 2021-03-16)
```

# Image Information

```bash
$ $ docker images
REPOSITORY           TAG                                     IMAGE ID       CREATED         SIZE
rustembedded/cross   powerpc-unknown-linux-gnu-0.2.1         e18a73ede16f   10 months ago   752MB
$ docker run -it e18a73ede16f /bin/bash
# lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.6 LTS
Release:        16.04
Codename:       xenial
# qemu-img --version
qemu-img version 3.0.1
Copyright (c) 2003-2017 Fabrice Bellard and the QEMU Project developers
```

The latest LTS version for Ubuntu is `20.04`, which has newer versions of important dependencies, such as Qemu.

```bash
$ docker run -it ubuntu:20.04 /bin/bash
# lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.2 LTS
Release:        20.04
Codename:       focal
# qemu-img --version
qemu-img version 4.2.1 (Debian 1:4.2-3ubuntu6.16)
Copyright (c) 2003-2019 Fabrice Bellard and the QEMU Project developers
```

# Issue

This actually matters, as numerous bugs, such as for float-point emulation, have been fixed in newer Qemu versions:
- [Inaccurate Fmod on i386](https://bugs.launchpad.net/qemu/+bug/1843205)

In addition, here's a sample code snippet that fails on Qemu 3.0.1 but works on Qemu 4.2.1:

```c
#include <inttypes.h>
#include <stdio.h>
#include <stdint.h>
#include <string.h>

double dn1() {
    double num = 1303.0;
    double den = 1e-20;
    return num * den;
}

double dn2() {
    double num = 1303.0;
    double den = 1e20;
    return num / den;
}


float fn1() {
    float num = 1303.0;
    float den = 1e-20;
    return num * den;
}

float fn2() {
    float num = 1303.0;
    float den = 1e20;
    return num / den;
}

int main() {
  float f = fn1();
  uint32_t bf;
  memcpy(&bf, &f, 8);
  printf("%u\n", bf);

  f = fn2();
  memcpy(&bf, &f, 8);
  printf("%u\n", bf);

  double d = dn1();
  uint64_t bd;
  memcpy(&bd, &d, 8);
  printf("%" PRIu64 "\n", bd);

  d = dn2();
  memcpy(&bd, &d, 8);
  printf("%" PRIu64 "\n", bd);

  return 0;
}
```

If we run the above using Qemu 3.0.1, we get:

```bash
# qemu-ppc --version
qemu-ppc version 3.0.1
Copyright (c) 2003-2017 Fabrice Bellard and the QEMU Project developers
# qemu-ppc ./main-ppc
594566255
594566255
4354430593920867240
4354430593920867241
```

If we run the above using Qemu 4.2.1, we get:

```bash
# qemu-ppc --version
qemu-ppc version 4.2.1 (qemu-4.2.1-1.fc32)
Copyright (c) 2003-2019 Fabrice Bellard and the QEMU Project developers
# qemu-ppc ./main-ppc
594566255
594566255
4354430593920867240
4354430593920867240
```

In short, we have a rounding error of 1 ULP due to cross using outdated Ubuntu (and therefore Qemu) versions. Ideally, we'd use version 6.0.1 or v5.0.0 (the latter requiring the following [PPA](https://launchpad.net/~jacob/+archive/ubuntu/virtualisation), the former requiring compiling from source).

**Comment by senden9** [comment]
Relates to #517

**Comment by Alexhuszagh** [comment]
The system packages for Focal (20.04) are still at 4.2.X, and v6.1.0 of Qemu has recently been released, and there's been numerous bug fixes in 5.X and 6.X. Since we already build from source, it should be trivial to update the local versions and enure all targets build.

**Comment by VictorKoenders** [comment]
I'm having an issue while running `powerpc64le-unknown-linux-gnu` where all floats are `0.0` and are not equal to each other 

![unknown](https://user-images.githubusercontent.com/2743142/163774821-8a6ebd6f-9963-41bd-8c1a-cfd25a2272d4.png)

Is this related or should I open a separate issue?

**Comment by Alexhuszagh** [comment]
@VictorKoenders How recent is this? It should be fixed in Qemu versions, although I'm not sure a cross-rs release has been published since then, as shown in https://github.com/cross-rs/cross/pull/586.

**Comment by VictorKoenders** [comment]
I think my issue is closer related to #644

**Comment by Emilgardis** [comment]
we're using 5.1.0 on upstream cross right now, try using that instead while we get ready for a release.

`cargo install cross --git https://github.com/cross-rs/cross`

or point the image you're using to the newest (unreleased) image

```toml
[target.{{target}}]
image = "ghcr.io/cross-rs/{{target}}:edge"
```

**Comment by Alexhuszagh** [comment]
@VictorKoenders They're related (I know from the Qemu bug tracker): the floating point implementation was somewhat bugged in older versions of Qemu, and pinning it to v3 (which I did) fixed some issues but causes other ones that existed in 4. In Qemu versions 5+, all of these are fixed.

**Comment by Alexhuszagh** [comment]
@Emilgardis Also this PR is ready but is waiting on the merge of https://github.com/cross-rs/cross/pull/593, since 2 of the images fail with interactive installed due to `tzdata`. Any review of that and this PR should be fit to be merged.
